storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java

```diff
@@ -305,8 +305,9 @@ private void setupFlushTupleTimer(final Map<String, Object> topologyConf, final
     }
 
     private void setupBackPressureCheckTimer(final Map<String, Object> topologyConf) {
-        final Integer workerCount = ObjectReader.getInt(topologyConf.get(Config.TOPOLOGY_WORKERS));
-        if (workerCount <= 1) {
+        Set<Integer> nonLocalTasks = Sets.difference(workerState.getTaskToComponent().keySet(),
```

Comment:
Perhaps we can consolidate these set difference calls to count single/multiworker mode into a single function ? 
- https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java#L230

Reply:
Some other places that may need similar fixes:

- https://github.com/apache/storm/blob/master/examples/storm-loadgen/src/main/java/org/apache/storm/loadgen/CaptureLoad.java#L109

- https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/StormCommon.java#L363-L364  

- https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/StormCommon.java#L276 


may want to check for all usages of that setting in code base.

Reply:
About the `set difference calls`, I thought these two are different. The first one https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/worker/Worker.java#L230 is to check if there is any outbound task not in local worker. This doesn't necessarily mean the topology has only one worker. 
The second (in setupBackPressureCheckTimer) is to check if all the tasks are in local which means there is only one worker.

Reply:
https://github.com/apache/storm/blob/master/examples/storm-loadgen/src/main/java/org/apache/storm/loadgen/CaptureLoad.java#L109 This is fine because the code also gets the num of workers from topologyInfo.

https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/StormCommon.java#L363-L364 this is also fine because we override the default value of `TOPOLOGY_EVENTLOGGER_EXECUTORS` when we submit the topology if it's on RAS.
Similar to https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/daemon/StormCommon.java#L276

Reply:
Thanks for clarifications. IMO, thats a nice (eventhough small) improvement to disable interworker when there are no outbound tasks... and not just for 1worker mode. 

Minor request: would be nice to have named functions for those two set difference calls. I suspect they will start seeing more usage soon.

looks good thanks for taking care of this.

Reply:
`Sets.difference()` is already a named function and it's almost one line of code. I think it's OK to just use it in this way?

Reply:
I meant ... would be nice to have functions like  isSingleWorkerMode() and hasOutboundTasks() that hide the detail of whatever mechanics are being employed to figure it out. I suspect they these functions will see more usage.